src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -99,16 +137,31 @@ public synchronized void close() throws IOException {
         Process killerProcess = new ProcessBuilder().command("taskkill", "/f", "/t", "/pid",
                 Integer.toString(Processes.newPidProcess(process).getPid())).start();
         try {
-            killerProcess.waitFor();
-            process.destroyForcibly();
-            process.waitFor(5, TimeUnit.SECONDS);
+            int taskkillExitCode = killerProcess.waitFor();
+            if (taskkillExitCode != 0) {
+                process.destroyForcibly();
+                process.waitFor(5, TimeUnit.SECONDS);
+            }
```

Comment:
this really shouldn't matter, we call destroyForcibly in the finally

Reply:
Gonna remove this one then... Not sure why we called it twice

Reply:
Keep it as it was. We want to use the waitFor to try and ensure that the process is as dead as possible when the function finishes.